nat-lab/tests/utils/testing.py

```diff
@@ -10,8 +12,102 @@ def unpack_optional(opt: Optional[T]) -> T:
     return opt
 
 
-def test_name_safe_for_file_name():
+def get_current_test_file_name() -> str | None:
+    """
+    Returns a string representing the filename of the currently running test.
+
+    The result is derived from PYTEST_CURRENT_TEST env variable.
+
+    For example:
+    "tests/test_mesh_plus_vpn.py" -> "test_mesh_plus_vpn"
+    """
+    test_name = os.environ.get("PYTEST_CURRENT_TEST")
+    if test_name is not None:
+        try:
+            # take the file path part of the pytest string, remove extension and parent directory
+            file_part = test_name.split("::")[0].split(".")[0].split("/")[-1]
+            return format_path_string(file_part)
+        except IndexError:
+            print(datetime.now(), "get_current_test_file_name() IndexError", test_name)
+            return test_name
+    print(datetime.now(), "PYTEST_CURRENT_TEST is None")
+    return None
+
+
+def get_current_test_case_name() -> str | None:
```

Comment:
I guess it would be more like `get_current_test_case_name_and_parameters() -> (str, str | None) | None` since some tests don't have parameters, though I like the idea of avoiding code duplication.

Reply:
I missed this but I think the signature can stay and empty string can be returned as param in such case. This way when you glue it togeter you can get rid of one if statement.

Reply:
`test_name, parameters = test_name.split("::", 1)[-1].split(" ")[0].split("[", 1)` actually crashes if there are no parameters

Reply:
This doesn't crash but I wouldn't use it ;)
```
 test_name, parameters, *_ = test_name.split("::", 1)[-1].split(" ")[0].split("[", 1) + [""]
```